src/main/java/org/mariadb/jdbc/internal/failover/impl/AuroraListener.java

```diff
@@ -114,6 +149,20 @@ public void reconnectFailedConnection(SearchFilter searchFilter) throws QueryExc
 
         resetOldsBlackListHosts();
 
+        // Initial connection to cluster - retrieve endpoints immediately
+        if (searchFilter.isInitialConnection() && urlParser.getHostAddresses().size() == 1 && getClusterHostAddress() != null) {
+            AuroraProtocol tempProtocol = AuroraProtocol.getNewProtocol(getProxy(), getUrlParser());
+            tempProtocol.setHostAddress(getClusterHostAddress());
+            try {
+                tempProtocol.connect();
+                retrieveAllEndpointsAndSet(tempProtocol);
+            } catch (QueryException e) {
```

Comment:
I sent you an email with an explanation for it :) Although it references "[...]if master is connected, no need to try a connection after ward, just set current master to this protocol"